Add template tag to enable react-refresh

BrandonShar opened the pull request:
This PR adds a template tag called `vite_react_refresh` to handle providing the extra code needed to support react hot module reloading. 

This was discussed a bit in #14 and #15 and while it is project specific and only relevant for react users, having this tag be part of the package is inline with other framework community adapters like [Vite Rails](https://vite-ruby.netlify.app/guide/troubleshooting.html#hot-module-refresh-does-not-work-for-react) and [Vite Laravel](https://laravel.com/docs/9.x/vite#react)

I wasn't sure how'd you like to format the multi-line string needed, but since it's dev only I figured the indenting didn't matter much. If you disagree, let me know and I'll revise it!

Thanks for this great repo!

MrBin99 commented:
Thanks for your PR but your code generates something for a specific frontend framework.

`django-vite` aims to be agnostic on the frontend framework you choose and if we start to implement every specificity for each frontend framework it will be very tedious.

I think `django-vite` must be very configurable to implement this by projects.

BrandonShar commented:
I’m a little confused, didn’t you say in #14 that you would accept a PR with this functionality? 

> if we start to implement every specificity for each frontend framework it will be very tedious.

Do you have an example of what you think this might lead to? I think react refresh is a special case because this is a known issue with Vite react that every other vite adapter I’ve seen handles. It’s more of a bug fix than an enhancement because it’s required for React to work with Vite.

MrBin99 commented:
Of course I will accept a PR, I didn't see that React is an official plugin of Vite so we could add it.
But I do not want to add to the project a lot of stuff like this for each frontend framework.
And I'm thinking a way to let the developer opt-in to use specific code for specific framework but there is no ideal way to do it.

So I think `django-vite` should support official plugins of Vite but that's all to not over complexify the lib.

Thanks for your PR, can you just make it more configurable and add some comments ? And I will merge your PR.

BrandonShar commented:
I'm happy to make changes, but I'm genuinely not sure what you're looking for. Both methods I added already have doc-blocks and there isn't much to configure. I added a setting for users to override the react-refresh url, but I don't even think that's necessary. Non-react users simply won't add the template tag, so there's nothing more that needs to be done there.

Let me know if there's something specific you're looking for!

thijskramer reviewed (approved):
Looks good to me! @MrBin99 would you please take another look?

AlessandroSalvetti commented:
Same for me, looks good. Could you @MrBin99 review it please?

BrandonShar commented:
I suspect this project has been abandoned. Since I maintain a project that's downstream of this, I think I'm going to fork it for a re-release. For those of you commenting here, is there anywhere (maybe discord?) that this project is discussed I could join?

thijskramer commented:
A fork that keeps getting maintained would be very nice! If you need any help with the maintenance, I'm happy to do help with that.  

I'm not sure if there is a space for discussions anywhere, maybe start a discussion in this repository? Or submit an issue with the proposal to possibly take over this project? Because it would be nice to keep the name 'django-vite'.

AlessandroSalvetti commented:
I can help too. 
If this project is abandoned and we need a fork, I think we should ask to Vite to be added in the official integration list ([here](https://github.com/vitejs/awesome-vite#django)) as this project is.

thijskramer commented:
Regarding #70 , @BrandonShar @AlessandroSalvetti and possibly others, would you be interested in a collaboration to revitalize this project? I've created a [Discord channel]https://discord.gg/P64Yk4qY() so we could discuss how to continue.

thijskramer commented:
Since I'm a It still looks good to me so I'll merge this!
